src/app/services/code-view/angular-code-service.ts

```diff
@@ -213,30 +240,47 @@ export class AngularCodeService extends CodeService {
         return demoFileMetadataPath;
     }
 
+    private createButtonForm(sampleContent: { [url: string]: any }, codeService: any, $button: JQuery<HTMLButtonElement>) {
```

Comment:
Should we have the codeservice as a parameter ? Isn't the context of this function the Codeservice itself?